test/e2e/rebalance-recommendation-drain-test

```diff
@@ -0,0 +1,166 @@
+#!/bin/bash
+set -euo pipefail
+
+# Available env vars:
+#   $TMP_DIR
+#   $CLUSTER_NAME
+#   $KUBECONFIG
+#   $NODE_TERMINATION_HANDLER_DOCKER_REPO
+#   $NODE_TERMINATION_HANDLER_DOCKER_TAG
+#   $WEBHOOK_DOCKER_REPO
+#   $WEBHOOK_DOCKER_TAG
+#   $AEMM_URL
+#   $AEMM_VERSION
+
+function fail_and_exit {
+    echo "❌ Rebalance Recommendation Drain test failed $CLUSTER_NAME ❌"
+    exit "${1:-1}"
+}
+
+echo "Starting Rebalance Recommendation Drain Test for Node Termination Handler"
+
+SCRIPTPATH="$( cd "$(dirname "$0")" ; pwd -P )"
+
+common_helm_args=()
+[[ "${TEST_WINDOWS-}" == "true" ]] && common_helm_args+=(--set targetNodeOs="windows")
+[[ -n "${NTH_WORKER_LABEL-}" ]] && common_helm_args+=(--set nodeSelector."$NTH_WORKER_LABEL")
+
+anth_helm_args=(
+  upgrade
+  --install
+  "$CLUSTER_NAME-anth"
+  "$SCRIPTPATH/../../config/helm/aws-node-termination-handler/"
+  --wait
+  --force
+  --namespace kube-system
+  --set instanceMetadataURL="${INSTANCE_METADATA_URL:-"http://$AEMM_URL:$IMDS_PORT"}"
+  --set image.repository="$NODE_TERMINATION_HANDLER_DOCKER_REPO"
+  --set image.tag="$NODE_TERMINATION_HANDLER_DOCKER_TAG"
+  --set enableScheduledEventDraining="false"
+  --set enableSpotInterruptionDraining="false"
+  --set enableRebalanceMonitoring="true"
```

Comment:
No code change needed, just want to confirm: it's not required to set both `enableRebalanceMonitoring` and `enableRebalanceDraining` to drain on Rebalance events, right? Only the latter?

Reply:
No, both are needed. You need the former to enable the monitor [here](https://github.com/aws/aws-node-termination-handler/blob/main/cmd/node-termination-handler.go#L148) and the latter to drain.

IMO it'd be better to follow the other monitors' approach, that is:
* Use `enableRebalanceDraining` param instead of `enableRebalanceMonitoring`
* Drain by default if the monitor is enabled, unless `cordonOnly == true`

This would be consistent with the existing code, but would also mean introducing a breaking change (new parameter, old gets deleted) and also a change in the default behaviour, so I opted for using the existing parameter to enable the monitor and adding a new parameter for enabling the drain.

Again, this and other things call for a bigger refactor that I think should happen some day. That would be the moment to introduce breaking changes, config changes and a major version bump IMHO.

Reply:
Also, haven't been able to test it using SQS. The actual code change is in the `drainOrCordonIfNecessary()` func, where an interruption event is passed in. I'm assuming such interruption event would get there just like it does now, regardless of what the source of the event (IMDB or SQS) is.

Reply:
1. Got it. I agree the current design and param names are a bit awkward as-is-- our team is working on a refactor to help address these issues. One way to bridge the gap now without introducing breaking changes would be to update the enable-the-monitor code you linked to `nthConfig.EnableRebalanceMonitoring || nthConfig.EnableRebalanceDraining` as it's safe to assume if you want draining then you want to monitor for that event. Adds a bit more tech debt for now, but I think it improves the experience-- could you make this change?

2. Your assumptions are correct, I'm just trying to validate if `enableRebalanceDraining` needs to be set to true in order to drain on Rebalance in QueueProcessor mode, which I think is true and that's what we want so we don't break the current default behavior. So all good here.

Reply:
Right, no need to check if the event is a rebalance recommendation. Just checking whether drain on rebalance is enabled is enough. Updating it.

Reply:
We could add something like
```
	if config.EnableRebalanceDraining {
		config.EnableRebalanceMonitoring = true
	}
```
in `pkg/config/config.go` to spare us from having to set both params. WDYT?

Reply:
I would rather have it more explicit. Instead could you update the enable monitoring logic [here](https://github.com/aws/aws-node-termination-handler/blob/main/cmd/node-termination-handler.go#L148) to be 
`nthConfig.EnableRebalanceMonitoring || nthConfig.EnableRebalanceDraining` ? It should result in the same behavior (not needing to set both params).

Reply:
Right, https://github.com/aws/aws-node-termination-handler/pull/400/commits/e99d19aa1bf53a495d1720af4b0ae25bfb9e297f should do it